Other/MergeShops/SunlightZoneHouseMerge.cs

```diff
@@ -0,0 +1,102 @@
+    private CoreAdvanced Adv = new();
+    private static CoreAdvanced sAdv = new();
+
+    private CoreSoW SoW = new();
+
+    public CoreAOR AOR = new();
+
+    public List<IOption> Generic = sAdv.MergeOptions;
+    public string[] MultiOptions = { "Generic", "Select" };
+    public string OptionsStorage = sAdv.OptionsStorage;
+    // [Can Change] This should only be changed by the author.
+    //              If true, it will not stop the script if the default case triggers and the user chose to only get mats
+    private bool dontStopMissingIng = false;
+
+    public void ScriptMain(IScriptInterface Bot)
+    {
+        Core.BankingBlackList.AddRange(new[] { "Sun Zone Chit", "Undine Visitor Badge"});
+        Core.SetOptions();
+
+        BuyAllMerge();
+        Core.SetOptions(false);
+    }
+
+    public void BuyAllMerge(string? buyOnlyThis = null, mergeOptionsEnum? buyMode = null)
+    {
+         AOR.SunlightZone();
+        //Only edit the map and shopID here
+        Adv.StartBuyAllMerge("sunlightzone", 2289, findIngredients, buyOnlyThis, buyMode: buyMode);
+
+        #region Dont edit this part
+        void findIngredients()
+        {
+            ItemBase req = Adv.externalItem;
+            int quant = Adv.externalQuant;
+            int currentQuant = req.Temp ? Bot.TempInv.GetQuantity(req.Name) : Bot.Inventory.GetQuantity(req.Name);
+            if (req == null)
+            {
+                Core.Logger("req is NULL");
+                return;
+            }
+
+            switch (req.Name)
+            {
+                default:
+                    bool shouldStop = Adv.matsOnly ? !dontStopMissingIng : true;
+                    Core.Logger($"The bot hasn't been taught how to get {req.Name}." + (shouldStop ? " Please report the issue." : " Skipping"), messageBox: shouldStop, stopBot: shouldStop);
+                    break;
+                #endregion
+
+                case "Sun Zone Chit":
+                    Core.FarmingLogger(req.Name, quant);
+                    Core.RegisterQuests(9251);
+                    while (!Bot.ShouldExit && !Core.CheckInventory(req.Name, quant))
+                    {
+                        Core.HuntMonster("sunlightzone", "Marine Snow", "Marine Sample");
+                        Core.HuntMonster("sunlightzone", "Infernal Illusion", "Infernal Sample", 10);
+                        Core.HuntMonster("sunlightzone", "Seraphic Illusion", "Seraphic Sample", 10);
+                        Bot.Wait.ForPickup(req.Name);
+                    }
+                    Core.CancelRegisteredQuests();
+                    break;
+
+                case "Undine Visitor Badge":
+                    Core.FarmingLogger(req.Name, quant);
+                    while (!Bot.ShouldExit && !Core.CheckInventory(req.Name, quant))
+                    {
+                        Core.HuntMonster("sunlightzone", "Spectral Jellyfish", req.Name, quant);
+                        Bot.Wait.ForPickup(req.Name);

```

Comment:
No need for while and pickup here, just use `Core.HuntMonster("sunlightzone", "Spectral Jellyfish", req.Name, quant, false);`